stories/util.js

```diff
@@ -50,7 +51,10 @@ Hits.propTypes = {
 export const CustomHits = connectHits(Hits);
 
 export const Wrap = ({ appId, apiKey, indexName, children }) => (
-  <InstantSearch appId={appId} apiKey={apiKey} indexName={indexName}>
+  <InstantSearch
+    searchClient={algoliasearch(appId, apiKey)}
```

Comment:
This can be stored in the React component otherwise it creates a new search client at every render.

Reply:
I've just tried it, and these components never rerender, so it's equivalent to storing it on state or something. 

Just because it's basically free I wrapped it in `useMemo`, but it still will be recalculated every render; because the whole tree is different & it wasn't kept.

<details>
<summary> state implementation </summary>

```js
export class WrapWithHits extends React.Component {
  state = {
    searchClient: algoliasearch(this.props.appId, this.props.apiKey),
    appId: this.props.appId,
    apiKey: this.props.apiKey,
  };

  static getDerivedStateFromProps(props, state) {
    if (props.appId !== state.appId || props.apiKey !== state.apiKey) {
      console.warn('recalcul');
      return {
        searchClient: algoliasearch(props.appId, props.apiKey),
        appId: props.appId,
        apiKey: props.apiKey,
      };
    }
    return {};
  }
  render() {
    const {
      searchParameters: askedSearchParameters = {},
      children,
      searchBox = true,
      hasPlayground = false,
      linkedStoryGroup,
      pagination = true,
      appId,
      apiKey,
      indexName,
      hitsElement,
    } = this.props;
    const { searchClient } = this.state;

    const sourceCodeUrl = `https://github.com/algolia/react-instantsearch/tree/master/stories/${linkedStoryGroup}.stories.js`;
    const playgroundLink = hasPlayground ? (
      <button
        onClick={linkTo(linkedStoryGroup, 'playground')}
        className="playground-url"
      >
        <span>Play with props</span>
      </button>
    ) : null;

    const footer = linkedStoryGroup ? (
      <div className="footer-container">
        {playgroundLink}
        <a
          href={sourceCodeUrl}
          className="source-code-url"
          target="_blank"
          rel="noopener noreferrer"
        >
          <div>View source code</div>
        </a>
      </div>
    ) : null;

    const hits = hitsElement || <CustomHits />;

    const searchParameters = {
      hitsPerPage: 3,
      ...askedSearchParameters,
    };

    return (
      <InstantSearch searchClient={searchClient} indexName={indexName}>
        <Configure {...searchParameters} />
        <div>
          <div className="container widget-container">{children}</div>
          <div>
            <div
              style={
                linkedStoryGroup ? {} : { borderRadius: '0px 0px 5px 5px' }
              }
              className="container hits-container"
            >
              <div className="hit-actions">
                {searchBox ? (
                  <SearchBox
                    translations={{
                      placeholder: 'Search into our products: phones, tv...',
                    }}
                  />
                ) : null}
                <ClearRefinements
                  translations={{ reset: 'Clear all filters' }}
                />
              </div>
              {hits}
              <div className="hit-pagination">
                {pagination ? <Pagination showLast={true} /> : null}
              </div>
            </div>
            {footer}
          </div>
        </div>
      </InstantSearch>
    );
  }
}
```
</details>